app/webpacker/components/Tickets/index.jsx

```diff
@@ -8,19 +10,91 @@ import TicketHeader from './TicketHeader';
 import TicketWorkbench from './TicketWorkbench';
 import TicketLogs from './TicketLogs';
 
+function SkateholderSelector({ stakeholderList, setUserSelectedStakeholder }) {
+  const [selectedOption, setSelectedOption] = useState(stakeholderList[0]);
+  return (
+    <>
+      <p>
+        You are part of more than one stakeholder, please select the stakeholder as which you want
+        to visit the ticket page.
+      </p>
+      <Dropdown
+        options={stakeholderList.map((requesterStakeholder) => ({
+          key: requesterStakeholder.id,
+          text: requesterStakeholder.stakeholder.name,
+          value: requesterStakeholder,
+        }))}
+        value={selectedOption}
+        onChange={(__, { value }) => setSelectedOption(value)}
+      />
+      <Button
+        disabled={!selectedOption}
+        onClick={() => setUserSelectedStakeholder(selectedOption)}
+      >
+        Select
+      </Button>
+    </>
+  );
+}
+
+function TicketContent({ ticketDetails, currentStakeholder, sync }) {
+  return (
+    <>
+      <TicketHeader ticketDetails={ticketDetails} />
+      <TicketWorkbench
+        ticketDetails={ticketDetails}
+        sync={sync}
+        currentStakeholder={currentStakeholder}
+      />
+      <TicketLogs logs={ticketDetails.ticket.ticket_logs} />
+    </>
+  );
+}
+
 export default function Tickets({ id }) {
   const {
     data: ticketDetails, sync, loading, error,
   } = useLoadedData(actionUrls.tickets.show(id));
 
+  const [userSelectedStakeholder, setUserSelectedStakeholder] = useState();
+  const currentStakeholder = userSelectedStakeholder || ticketDetails?.requester_stakeholders[0];
+  const shouldUserSelectStakeholder = (
+    !userSelectedStakeholder
+    && ticketDetails?.requester_stakeholders?.length > 1);
+
   if (loading) return <Loading />;
   if (error) return <Errored />;
 
   return (
-    <Container fluid>
-      <TicketHeader ticketDetails={ticketDetails} />
-      <TicketWorkbench ticketDetails={ticketDetails} sync={sync} />
-      <TicketLogs logs={ticketDetails.ticket.ticket_logs} />
-    </Container>
+    <>
+      <Container fluid>
+        {userSelectedStakeholder && (
+          <Message>
+            {`You are currently viewing the ticket as stakeholder "${userSelectedStakeholder.stakeholder.name}".`}
+            <Button
+              onClick={() => setUserSelectedStakeholder(false)}
+            >
+              Click here to change
+            </Button>
+          </Message>
+        )}
+        {!shouldUserSelectStakeholder && (
```

Comment:
It was very hard for me to figure out during review why you only show the ticket body if the user doesn't have to select a stakeholder.
Actually it's more like "When the user has selected a stakeholder or when we can derive the stakeholder automatically". Is there any way you can come up with a better boolean combination + clearer naming?

Reply:
For example, you could make it so that `requester_stakeholder[0]` is only used as default if there is only one stakeholder in the first place. Then this check would be as easy as `currentStakeholder && (<TicketContent>)` which is far more readable

Reply:
Okay, accepted your suggestion. Hope this is more readable now. Please let me know if any more changes are required.